arviz/data/inference_data.py

```diff
@@ -324,6 +381,325 @@ def sel(
+        """Perform xarray renaming of dimensions on all groups.
+
+        Loops groups to perform Dataset.rename_dims(name_dict)
+        for every key in name_dict if key is a dimension of the dataset.
+        The renaming is performed on all relevant groups (like
+        posterior, prior, sample stats) while non relevant groups like observed data are
+        omitted. See :meth:`xarray:xarray.Dataset.rename_dims`
+
+        Parameters
+        ----------
+        name_dict: dict
+            Dictionary whose keys are current dimension names and whose values are the desired
+            names.
+        groups: str or list of str, optional
+            Groups where the selection is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        inplace: bool, optional
+            If ``True``, modify the InferenceData object inplace,
+            otherwise, return the modified copy.
+
+
+        Returns
+        -------
+        InferenceData
+            A new InferenceData object with renamed dimension by default.
+            When `inplace==True` perform renaming in-place and return `None`
+
+        """
+
+        groups = self._group_names(groups, filter_groups)
+        if "chain" in name_dict.keys() or "draw" in name_dict.keys():
+            raise KeyError("'chain' or 'draw' dimensions can't be renamed")
+
+        out = self if inplace else deepcopy(self)
+        for group in groups:
+            dataset = getattr(self, group)
+            valid_keys = set(name_dict.keys()).intersection(dataset.dims)
+            dataset = dataset.rename_dims({key: name_dict[key] for key in valid_keys})
+            setattr(out, group, dataset)
+        if inplace:
+            return None
+        else:
+            return out
+
+    set_index = _extend_xr_method(xr.Dataset.set_index)
+    get_index = _extend_xr_method(xr.Dataset.get_index)
+    reset_index = _extend_xr_method(xr.Dataset.reset_index)
+    set_coords = _extend_xr_method(xr.Dataset.set_coords)
+    reset_coords = _extend_xr_method(xr.Dataset.reset_coords)
+    assign = _extend_xr_method(xr.Dataset.assign)
+    assign_coords = _extend_xr_method(xr.Dataset.assign_coords)
+    sortby = _extend_xr_method(xr.Dataset.sortby)
+
+    mean = _extend_xr_method(xr.Dataset.mean)
+    median = _extend_xr_method(xr.Dataset.median)
+    min = _extend_xr_method(xr.Dataset.min)
+    max = _extend_xr_method(xr.Dataset.max)
+    cumsum = _extend_xr_method(xr.Dataset.cumsum)
+    sum = _extend_xr_method(xr.Dataset.sum)
+    quantile = _extend_xr_method(xr.Dataset.quantile)
```

Comment:
and these ones would become `self._extend...`

Reply:
this is returning error `self not defined`